### PR TITLE
fix(direktverbindung): display the main layer when day or night layer become visible

### PR DESCRIPTION
**Pull request by oterral**
# How to

This PR contains 2 things:
- update from some dependencies but mainly react-spatial and its new canvas save button. So we must test every export button : tarifverbunkarte, Gektunsbereichex2 and the default export button in share menu
- fix to display the dv main layer when dv night and day are not visible on load  [test link](https://dev-trafimage-maps-git-olivier-trafwart-1650-geops.vercel.app/ch.sbb.direktverbindungen?lang=de&x=928460&y=5908948&z=8.5&layers=&baselayers=ch.sbb.direktverbindungen.base-light,ch.sbb.direktverbindungen.base-dark,ch.sbb.direktverbindungen.base-aerial), commit to review [here](https://github.com/geops/trafimage-maps/pull/1176/commits/c60da846a88898fe8736195e6e6ceb7301ab36dd)

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being and follows the [conventional commits](https://www.conventionalcommits.org/) specification.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [x] Tests added.

### CODE REVIEW

**Review by danji90 — CHANGES_REQUESTED** [review]
Looks good, but I found some more things to fix:
- Tarifverbundkarte cypress tests are failing, while testing I noticed some console errors and the highlighting on the map is broken (the zones do not show in red on click). I guess it has to do with the turf upgrade
- https://dev-trafimage-maps-git-olivier-trafwart-1650-geops.vercel.app/ch.sbb.sts?lang=de&layers=ch.sbb.direktverbindungen.night&baselayers=ch.sbb.direktverbindungen.data,ch.sbb.sts.validity.data should switch on night trains only on load, but it always switches on both in the end. This needs to be fixed in the StsMenu, so maybe in a different ticket since it's not directly related.

**Comment by oterral** [comment]
@danji90  can you re-review please

**Review by danji90 — APPROVED** [review]
@oterral Typo in the comment, otherwise LGTM

File: src/layers/TarifverbundkarteLayer/TarifverbundkarteLayer.js [hunk 1]
```diff
@@ -91,37 +91,19 @@ class TarifverbundkarteLayer extends MapboxStyleLayer {
...
+      if (!feature || !zones?.length) {
         // Abort if no zone or feature present
         return null;
       }
 
       /**
-       * If multiple zones present, the zones are intersected with one another
-       * Then they are intersected with the municipality feature separately for readability
-       * @ignore
+       * If multiple zones present, the zones are intersected with one another adn the municipality feature
```
[the remark this answers is not in the document]
Comment (oterral) [reply src/layers/TarifverbundkarteLayer/TarifverbundkarteLayer.js]: fixed

**Pull request merged**